src/ui/public/kuery/lib/functions/range.js

```diff
@@ -0,0 +1,79 @@
+import _ from 'lodash';
+import { nodeTypes } from '../node_types';
+import * as ast from '../ast';
+import { getRangeScript } from 'ui/filter_manager/lib/range';
+
+export function buildNodeParams(fieldName, params, serializeStyle = 'shorthand') {
+  params = _.pick(params, 'gt', 'lt', 'gte', 'lte');
+  const fieldNameArg = nodeTypes.literal.buildNode(fieldName);
+  const args = _.pairs(params).map((argument) => {
+    const [ name, value ] = argument;
+    return nodeTypes.namedArg.buildNode(name, value);
+  });
+
+  // we only support inclusive ranges in the shorthand syntax currently
+  if (_.has(params, 'gt') || _.has(params, 'lt')) {
+    serializeStyle = 'function';
+  }
+
+  return {
+    arguments: [fieldNameArg, ...args],
+    serializeStyle,
+  };
+}
+
+export function toElasticsearchQuery(node, indexPattern) {
+  const [ fieldNameArg, ...args ] = node.arguments;
+  const fieldName = ast.toElasticsearchQuery(fieldNameArg);
```

Comment:
In other places the `fieldName` is hardcoded to be a `literal`. Is this case different on purpose?